app/models/ability.rb

```diff
@@ -205,62 +238,66 @@ def public_ability
     can :index, CrossroadsTransport
   end
 
-  def address_abilities
-    # User address
-    can [:create, :show], Address, addressable_type: "User", addressable_id: @user_id
-
-    # Offer delivery address
-    can [:create, :show, :destroy], Address, addressable_type: "Contact", addressable: { delivery: { offer_id: @user_offer_ids } }
-    can [:create, :show, :destroy], Address, addressable_type: "Contact" if staff?
+  def packages_locations_abilities
+    if can_access_packages_locations? || @api_user
+      can [:index, :show], PackagesLocation
+    end
   end
 
-  def schedule_abilities
-    can [:create, :availableTimeSlots], Schedule
-    can [:index, :show], Schedule, deliveries: { offer_id: @user_offer_ids }
-    can [:index, :show], Schedule if staff?
+  def stockit_organisation_abilities
+    can [:create, :index], StockitOrganisation if @api_user
   end
 
-  def gogovan_order_abilities
-    can [:calculate_price, :confirm_order, :destroy], GogovanOrder, delivery: { offer_id: @user_offer_ids }
-    can [:calculate_price, :confirm_order, :destroy], GogovanOrder if staff?
+  def stockit_contact_abilities
+    can [:create, :index], StockitContact if @api_user
   end
 
-  def contact_abilities
-    can :destroy, Contact, delivery: { offer_id: @user_offer_ids }
-    can :destroy, Contact if staff?
-    can :create, Contact
+  def stockit_local_order_abilities
+    can [:create, :index], StockitLocalOrder if @api_user
   end
 
   def taxonomies
     can :register, :device
     can [:index, :show], DonorCondition
     can [:index, :show], SubpackageType
     can [:index, :show], RejectionReason
+    can [:index, :show], Role
     can [:index, :show], Permission
+    can [:index, :show], UserRole
     can [:index, :show], CancellationReason
-    can :create, PackageType if @api_user || staff?
-    can [:create, :remove_number], InventoryNumber if api_user_or_staff?
+    if can_add_or_remove_inventory_number? || @api_user
+      can [:create, :remove_number], InventoryNumber
+    end
+  end
+
+  def schedule_abilities
+    can [:create, :availableTimeSlots], Schedule
+    can [:index, :show], Schedule, deliveries: { offer_id: @user_offer_ids }
+    can [:index, :show], Schedule if can_read_schedule?
+  end
+
+  def stockit_abilities
+    if can_manage_locations? || @api_user
+      can [:index, :create, :destroy], Location
+    end
+
+    if @api_user
+      can [:create, :index], Box
+      can [:create, :index], Pallet
+      can [:create, :index], Country
+      can [:create, :index], StockitActivity
+    end
   end
 
   def user_abilities
-    can [:show, :update], User, id: @user_id
-    can [:index, :show, :update], User if staff?
     can :current_user_profile, User
+    can [:show, :update], User, id: @user_id
+    can [:index, :show, :update], User if can_read_or_modify_user?
```

Comment:
Can we make this `can_read_or_update_users?` 'Modify' is fine but we already use 'update'